ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -23,19 +23,44 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
-   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  async model({ target_id, isConnecting }) {
-    const target = await this.store.findRecord('target', target_id, {
+  model({ target_id }) {
+    return this.store.findRecord('target', target_id, {
       reload: true,
     });
+  }
 
-    if (isConnecting) {
-      await this.preConnect(target);
-    }
+  async afterModel(model, transition) {
```

Comment:
I actually feel most of this fetching should happen in the `model` hook which is where it seems most appropriate. When i had originally mentioned you could use the `afterModel` hook it was specifically for when you check for if you should directly connect (lines 58-60) since that causes a transition and that makes sense to have it in the afterModel hook. That way the actual model returned from the `model` hook is accurate

Reply:
Gotcha. That makes sense. I will update that now.